mixer/adapter/fluentd/fluentd.go

```diff
@@ -0,0 +1,162 @@
+	PostWithTime(string, time.Time, interface{}) error
+}
+
+// ensure types implement the requisite interfaces
+var _ logentry.HandlerBuilder = &builder{}
+var _ logentry.Handler = &handler{}
+
+///////////////// Configuration-time Methods ///////////////
+
+// adapter.HandlerBuilder#Build
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+	h, pStr, err := net.SplitHostPort(b.adpCfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	p, err := strconv.Atoi(pStr)
+	if err != nil {
+		return nil, err
+	}
+	han := &handler{
+		types: b.types,
+		env:   env,
+	}
+	han.logger, err = fluent.New(fluent.Config{FluentPort: p, FluentHost: h})
+	if err != nil {
+		return nil, err
+	}
+	return han, nil
+}
+
+func (b *builder) injectBuild(ctx context.Context, env adapter.Env, l fluentdLogger) (adapter.Handler, error) {
+	han := &handler{
+		logger: l,
+		types:  b.types,
+		env:    env,
+	}
+	return han, nil
+}
+
+// adapter.HandlerBuilder#SetAdapterConfig
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.adpCfg = cfg.(*config.Params)
+}
+
+// adapter.HandlerBuilder#Validate
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	if b.adpCfg.Address == "" {
+		ce = ce.Appendf("address", "Address is empty")
+	}
+	if _, _, err := net.SplitHostPort(b.adpCfg.Address); err != nil {
+		ce = ce.Appendf("address", "Address is malformed: %v", err)
+	}
+	return
+}
+
+// logentry.HandlerBuilder#SetLogEntryTypes
+func (b *builder) SetLogEntryTypes(types map[string]*logentry.Type) {
+	b.types = types
+}
+
+////////////////// Request-time Methods //////////////////////////
+
+// logentry.Handler#HandleLogEntry
+func (h *handler) HandleLogEntry(ctx context.Context, insts []*logentry.Instance) error {
+	for _, i := range insts {
+		if h.env.Logger().VerbosityLevel(4) {
+			h.env.Logger().Infof("Got a new log for fluentd, name %v", i.Name)
+		}
+
+		// Durations are not supported by msgp
+		for k, v := range i.Variables {
+			if h.types[i.Name].Variables[k] == descriptor.DURATION {
+				d := v.(time.Duration)
+				i.Variables[k] = d.String()
+			}
+		}
+
+		i.Variables["severity"] = i.Severity
+
+		tag, ok := i.Variables["tag"]
+		if !ok {
+			tag = i.Name
+		} else {
+			i.Variables["name"] = i.Name
+			delete(i.Variables, "tag")
+		}
+
+		err := h.logger.PostWithTime(tag.(string), i.Timestamp, i.Variables)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// adapter.Handler#Close
+func (h *handler) Close() error {
+	return h.logger.Close()
+}
+
+////////////////// Bootstrap //////////////////////////
+
+// GetInfo returns the adapter.Info specific to this adapter.
+func GetInfo() adapter.Info {
+	return adapter.Info{
+		Name:        "fluentd",
+		Description: "Sends logentrys to a fluentd instance",
+		SupportedTemplates: []string{
+			logentry.TemplateName,
+		},
+		NewBuilder: func() adapter.HandlerBuilder { return &builder{} },
+		DefaultConfig: &config.Params{
+			Address: "localhost:24224",
```

Comment:
should we move this up into a const and make it more obvious as to the default value?

Reply:
Done